src/drafter.cc

```diff
@@ -1,22 +1,192 @@
-//
-//  drafter.cc
-//  drafter
-//
-//  Created by Jiri Kratochvil on 2015-03-05
-//  Copyright (c) 2015 Apiary Inc. All rights reserved.
-//
-
 #include "drafter.h"
 
-namespace drafter {
+#include "snowcrash.h"
+
+#include "refract/Element.h"
+#include "refract/FilterVisitor.h"
+#include "refract/Query.h"
+#include "refract/Iterate.h"
+
+
+#include "SerializeResult.h" // FIXME: remove - actualy required by WrapParseResultRefract()
+#include "Serialize.h" // FIXME: remove - actualy required by WrapperOptions
+#include "RefractDataStructure.h" // FIXME: remove - required by SerializeRefract()
+
+#include "sos.h" // FIXME: remove sos dependency
+#include "sosJSON.h"
+#include "sosYAML.h"
+
+#include "ConversionContext.h"
+
+#include "Version.h"
+
+#include <string.h>
+
+#include <assert.h>
+
+DRAFTER_API int drafter_parse_blueprint_to(const char* source,
+                                           char ** out,
+                                           const drafter_options options) {
+
+    assert(source);
+    assert(out);
+
+    if (!source || !out) {
+        return -1;
+    }
+
+    drafter_result* result = nullptr;
+    *out = nullptr;
+
+    int ret = drafter_parse_blueprint(source, &result);
+
+    assert(result);
+
+    if (!result) {
+        return -1;
+    }
+
+    *out = drafter_serialize(result, options);
+
+    drafter_free_result(result);
+
+    return ret;
+}
+
+namespace sc = snowcrash;
+
+/* Parse API Bleuprint and return result, which is a opaque handle for
+ * later use*/
+DRAFTER_API int drafter_parse_blueprint(const char* source, drafter_result** out) {
+
+    assert(source);
+    assert(out);
+
+    if (!source) {
```

Comment:
I agree, @klokane choose your style and keep it everywhere, using either assert or check for nullptr.